mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import threading
+import re
+from mongo_connector import errors
+
+
+LOG = logging.getLogger(__name__)
+
+"""New structure to handle both plain and wildcard mapping namespaces dynamically.
+"""
+
+class DestMapping():
+    def __init__(self, namespace_set=[], ex_namespace_set=[], inpt={}):
+        # a dict containing plain mappings
+        self.plain = {}
+        # a dict containing wildcard mappings
+        self.wildcard = {}
+        # a dict containing reverse plain mapping
+        # because the mappings are not duplicated, so the values should also be unique
+        self.reverse_plain = {}
+        # a dict containing plain db mappings, db -> an array of mapped db
+        self.plain_db = {}
+        
+        # the input namespace_set and ex_namespace_set could contain wildcard
+        self.namespace_set = namespace_set
+        self.ex_namespace_set = ex_namespace_set
+        
+        self.lock = threading.Lock()
```

Comment:
The DestMapping object will be changed when there is a new wildcard mapping used by get function. And this DestMapping will be passed to create OplogThread and could be multi-thread.  I never used this Lock feature before, please correct me if anything is wrong.